packages/lodestar/src/chain/chain.ts

```diff
@@ -345,54 +346,39 @@ export class BeaconChain extends (EventEmitter as {new (): ChainEventEmitter}) i
     this.logger.profile("restoreHeadState");
   }
 
-  private async initForkChoice(
-    lastKnownState: TreeBacked<BeaconState>,
-    finalizedBlock: SignedBeaconBlock
-  ): Promise<void> {
-    const isStateNotGenesis = lastKnownState.slot > GENESIS_SLOT;
-    const finalizedSlot = finalizedBlock.message.slot;
-    const blockCheckpoint: Checkpoint = {
-      root: this.config.types.BeaconBlock.hashTreeRoot(finalizedBlock.message),
-      epoch: computeEpochAtSlot(this.config, finalizedSlot),
-    };
-    // add justified block to forkchoice so "this.justified" in forkchoice really map to a block
-    if (isStateNotGenesis) {
-      const preJustifiedBlock = await this.db.blockArchive.getByRoot(lastKnownState.currentJustifiedCheckpoint.root);
-      let preFinalizedBlocks = await this.db.blockArchive.values({
-        gt: preJustifiedBlock!.message.slot,
-        lt: finalizedSlot,
-      });
-      preFinalizedBlocks = sortBlocks([preJustifiedBlock!, ...preFinalizedBlocks]);
-      const firstSlot = preFinalizedBlocks[0].message.slot;
-      const lastSlot = preFinalizedBlocks[preFinalizedBlocks.length - 1].message.slot;
-      this.logger.info(`Initialize forkchoice with pre-finalized blocks from ${firstSlot} to ${lastSlot}
-        and finalized block ${finalizedSlot}`);
-      preFinalizedBlocks.forEach((block) => {
-        this.forkChoice.addBlock({
-          slot: block.message.slot,
-          blockRoot: this.config.types.BeaconBlock.hashTreeRoot(block.message),
-          stateRoot: block.message.stateRoot.valueOf() as Uint8Array,
-          parentRoot: block.message.parentRoot.valueOf() as Uint8Array,
-          // don't care the below justified and finalized checkpoint as we don't use them
-          justifiedCheckpoint: {epoch: 0, root: ZERO_HASH},
-          finalizedCheckpoint: {epoch: 0, root: ZERO_HASH},
-        });
-      });
+  /**
+   * Seeds the fork choice with an anchor state.
+   * This state is set as the finalized state.
+   */
+  private async initForkChoice(anchorState: TreeBacked<BeaconState>): Promise<void> {
+    let blockRoot;
+    if (anchorState.latestBlockHeader.slot === GENESIS_SLOT) {
+      const block = getEmptyBlock();
+      block.stateRoot = this.config.types.BeaconState.hashTreeRoot(anchorState);
+      blockRoot = this.config.types.BeaconBlock.hashTreeRoot(block);
+    } else {
+      const blockHeader = anchorState.latestBlockHeader as BeaconBlockHeader;
+      blockHeader.stateRoot = this.config.types.BeaconState.hashTreeRoot(anchorState);
```

Comment:
maybe `anchorState.hashTreeRoot()`?

If I use `this.config.types.BeaconState.hashTreeRoot(anchorState)`, I got a wrong state root, do you know why?

Reply:
the two should be equivalent. If you can get a reproducible testcase theres a bug!

Reply:
thanks for figuring that out, that's my environment issue.
probably we should clone `latestBlockHeader` before assigning new `stateRoot` so that the original `anchorState` is unchanged? or just clone the original anchorState.